### PR TITLE
[perf] use as_json param of fhirstore.search to avoid validation

### PR DESCRIPTION
**Pull request by simonvadee**
## Fixes
 Validation was enabled on search results which greatly impacted the performance.

## Description
The `as_json` parameter allows to bypass Bundle validation by simply returning the Bundle document as plain JSON.

## Tests
Unit tests were missing for the search method

---
This change is [Reviewable](https://reviewable.io/reviews/arkhn/warehouse-api/156)

### CODE REVIEW

File: fhir-api/tests/models/test_resources.py [hunk 1]
```diff
@@ -223,12 +220,73 @@ def test_delete_missing_id(self, mock_get_store):
...
         with pytest.raises(
-            BadRequest,
-            match="Resource ID is required to delete it",
+            BadRequest, match="Resource ID is required to delete it",
         ):
             r = r.delete()
         assert mock_get_store.return_value.update.call_count == 0
 
+    def test_search_qs(self, mock_get_store):
+        """Calls the delete method of the fhirstore client"""
```
Comment (Jasopaum) [inline fhir-api/tests/models/test_resources.py:229]: You forgot to change the doc on the new tests

File: fhir-api/tests/models/test_resources.py [hunk 2]
```diff
@@ -223,12 +220,73 @@ def test_delete_missing_id(self, mock_get_store):
...
+        mock_get_store.return_value.normalize_resource.return_value = resource
+        r = BaseResource(resource=resource)
+
+        r.search(query_string="name=Vincent")
+
+        mock_get_store.return_value.search.assert_called_once_with(
+            "BaseResource", as_json=False, params=None, query_string="name=Vincent"
+        )
+        assert r.resource == resource
+        assert r.id == resource.id
```
Comment (Jasopaum) [inline fhir-api/tests/models/test_resources.py:240]: I don't get these tests 🤔 
Does `r.search` modifies something in place?

File: fhir-api/tests/models/test_resources.py [hunk 3]
```diff
@@ -223,12 +220,73 @@ def test_delete_missing_id(self, mock_get_store):
...
+        params = MultiDict((("name", "Vincent"),))
+        r.search(params=params)
+
+        mock_get_store.return_value.search.assert_called_once_with(
+            "BaseResource", as_json=False, params=params, query_string=None
+        )
+        assert r.resource == resource
+        assert r.id == resource.id
+
+    def test_search_as_json(self, mock_get_store):
```
Comment (Jasopaum) [inline fhir-api/tests/models/test_resources.py:257]: We could also verify the output type here and/or that no validation was performed

File: fhir-api/fhir_api/models/base.py [hunk 4]
```diff
@@ -106,9 +106,13 @@ def delete(self) -> OperationOutcome:
         self.id = None
         return res
 
-    def search(self, query_string=None, params=None) -> Union[Bundle, OperationOutcome]:
+    def search(
+        self, query_string=None, params=None, as_json=False
```
Comment (Jasopaum) [inline fhir-api/fhir_api/models/base.py:110]: Will there be a use case where we want to perform a search with the validation enabled?
I think at least the default behavior should be `as_json=True`, right?

File: fhir-api/tests/models/test_resources.py (hunk 2 above)
Comment (simonvadee) [reply fhir-api/tests/models/test_resources.py:240]: no it does not, the resource must remain the same as model initialization

File: fhir-api/tests/models/test_resources.py (hunk 1 above)
Comment (simonvadee) [reply fhir-api/tests/models/test_resources.py:229]: done

File: fhir-api/tests/models/test_resources.py (hunk 3 above)
Comment (simonvadee) [reply fhir-api/tests/models/test_resources.py:257]: since fhirstore is mocked, it would be useless

File: fhir-api/fhir_api/models/base.py (hunk 4 above)
Comment (simonvadee) [reply fhir-api/fhir_api/models/base.py:110]: right, done

File: fhir-api/tests/models/test_resources.py [hunk 5]
```diff
@@ -249,27 +253,33 @@ def test_search_params(self, mock_get_store):
         r.search(params=params)
 
         mock_get_store.return_value.search.assert_called_once_with(
-            "BaseResource", as_json=False, params=params, query_string=None
+            "BaseResource", as_json=True, params=params, query_string=None
         )
         assert r.resource == resource
         assert r.id == resource.id
 
     def test_search_as_json(self, mock_get_store):
```
Comment (Jasopaum) [inline fhir-api/tests/models/test_resources.py:265]: `_as_resource`?

**Pull request merged**